### PR TITLE
use the symver attribute, if available, for versioned symbols,

### PR DESCRIPTION
**Pull request by sezero**
e.g. with gcc >= 10.
Co-authored-by: Claudio Matsuoka
Closes: https://github.com/libxmp/libxmp/issues/385

### CODE REVIEW

**Comment by cmatsuoka** [comment]
@sezero We ported Travis tests to GH actions so if you merge master into this branch it should stop complaining about Travis not responding.

**Comment by cmatsuoka** [comment]
And I think Dominik should review this PR before we land it.

**Comment by sezero** [comment]
> @sezero We ported Travis tests to GH actions so if you merge master into this branch it should stop complaining about Travis not responding.

Done.

**Comment by sezero** [comment]
> And I think Dominik should review this PR before we land it.

@rathann?

**Comment by sezero** [comment]
The gcc versions on the buildbots don't seem to recognize symver attribute..

**Comment by AliceLR** [comment]
> The gcc versions on the buildbots don't seem to recognize symver attribute..

I don't think Ubuntu has GCC 10+ by default—might be worth adding a Fedora regression test if possible.

**Comment by sezero** [comment]
Alice: the new buildbot with lto emits warnings:
```
test_depack_it_sample_16bit.c:4:5: warning: type of ‘itsex_decompress16’ does not match original declaration [-Wlto-type-mismatch]
    4 | int itsex_decompress16(HIO_HANDLE *module, void *dst, int len, char it215);
      |     ^
../src/loaders/itsex.c:131:5: note: type mismatch in parameter 4
  131 | int itsex_decompress16(HIO_HANDLE *src, int16 *dst, int len, int it215)
      |     ^
../src/loaders/itsex.c:131:5: note: type ‘int’ should match type ‘char’
../src/loaders/itsex.c:131:5: note: ‘itsex_decompress16’ was previously declared here

test_depack_it_sample_8bit.c:4:5: warning: type of ‘itsex_decompress8’ does not match original declaration [-Wlto-type-mismatch]
    4 | int itsex_decompress8(HIO_HANDLE *module, void *dst, int len, char it215);
      |     ^
../src/loaders/itsex.c:36:5: note: type mismatch in parameter 4
   36 | int itsex_decompress8(HIO_HANDLE *src, uint8 *dst, int len, int it215)
      |     ^
../src/loaders/itsex.c:36:5: note: type ‘int’ should match type ‘char’
../src/loaders/itsex.c:36:5: note: ‘itsex_decompress8’ was previously declared here
```

**Comment by AliceLR** [comment]
> Alice: the new buildbot with lto emits warnings:

Seems to be unrelated to this branch, I just missed it before—the declarations in the IT sample unpacking tests are wrong. I'll submit a separate patch for it.

**Review by rathann — COMMENTED** [review]
Works for me on Fedora 34. Regression tests pass and xmp plays a few random modules I have around.

File: lite/configure.ac
Comment (rathann) [inline lite/configure.ac:141]: This seems unrelated, but I guess this was done for consistency of `configure` output. I'd make it a separate commit, though.

**Comment by sezero** [comment]
@cmatsuoka what is your decision?

File: lite/configure.ac
Comment (sezero) [reply lite/configure.ac]: removed that cosmetic change for now, will push when the buildbots complete.

**Comment by sezero** [comment]
@AliceLR or @cmatsuoka: is this OK? (need a final review.)

**Review by AliceLR — APPROVED** [review]
I have limited knowledge with regard to versioned symbols, but this looks OK to me and configures/compiles fine with GCC 11.1.1/-flto in my Fedora 34 VM. Approving.

**Pull request merged**
